runelite-client/src/main/java/net/runelite/client/plugins/statusbars/BarRenderer.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (c) 2018, Rheon <https://github.com/Rheon-D>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *     list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *     this list of conditions and the following disclaimer in the documentation
+ *     and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+package net.runelite.client.plugins.statusbars;
+
+import lombok.RequiredArgsConstructor;
+import static net.runelite.client.plugins.statusbars.StatusBarsOverlay.BACKGROUND;
+import static net.runelite.client.plugins.statusbars.StatusBarsOverlay.COUNTER_ICON_HEIGHT;
+import static net.runelite.client.plugins.statusbars.StatusBarsOverlay.HEAL_OFFSET;
+import static net.runelite.client.plugins.statusbars.StatusBarsOverlay.ICON_AND_COUNTER_OFFSET_X;
+import static net.runelite.client.plugins.statusbars.StatusBarsOverlay.ICON_AND_COUNTER_OFFSET_Y;
+import static net.runelite.client.plugins.statusbars.StatusBarsOverlay.OFFSET;
+import static net.runelite.client.plugins.statusbars.StatusBarsOverlay.OVERHEAL_COLOR;
+import static net.runelite.client.plugins.statusbars.StatusBarsOverlay.OVERHEAL_OFFSET;
+import static net.runelite.client.plugins.statusbars.StatusBarsOverlay.PADDING;
+import static net.runelite.client.plugins.statusbars.StatusBarsOverlay.SKILL_ICON_HEIGHT;
+import static net.runelite.client.plugins.statusbars.StatusBarsOverlay.WIDTH;
+import net.runelite.client.ui.FontManager;
+import net.runelite.client.ui.overlay.components.TextComponent;
+
+import java.awt.Color;
+import java.awt.Graphics2D;
+import java.awt.Image;
+import java.util.function.Supplier;
+
+@RequiredArgsConstructor
+class BarRenderer
+{
+	private static final TextComponent textComponent = new TextComponent();
```

Comment:
Naming scheme should be `TEXT_COMPONENT` as it's a static final variable.

Reply:
why does this exist